### PR TITLE
Add dockerfile to build multipy

### PR DESCRIPTION
**Pull request by PaliC**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #110
* __->__ #107

Adds a dockerfile to help build multipy. In order to run it you need to have the nvidia-docker runtime installed which is described [https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/install-guide.html#docker](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/install-guide.html#docker)

This is fairly useful for having a GPU build.

Afterwards go to the multipy root directory it's just 
```
git submodule update --init --recursive
DOCKER_BUILDKIT=1 nvidia-docker build -t <tag name> .
```

Differential Revision: [D38430368](https://our.internmc.facebook.com/intern/diff/D38430368)

### CODE REVIEW

**Comment by PaliC** [comment]
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38430368).

**Comment by PaliC** [comment]
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38430368).

**Comment by PaliC** [comment]
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38430368).

**Comment by PaliC** [comment]
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38430368).

**Comment by PaliC** [comment]
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38430368).

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,123 @@
+ARG BASE_IMAGE=nvidia/cuda:11.3.1-devel-ubuntu18.04
+ARG PYTHON_VERSION=3.8
+
+FROM ${BASE_IMAGE} as dev-base
+
+# Install system dependencies
+RUN --mount=type=cache,id=apt-dev,target=/var/cache/apt \
```
Comment (kurman) [inline Dockerfile:7]: I wonder if there a image that we could use here instead of building it ourselves?

**Review by kurman — APPROVED** [review]

**Pull request closed without merging**